### ISSUE TITLE
Transition TransNetV2 from TensorFlow to PyTorch for CUDA Support

### ISSUE THREAD

**Issue by ssilverrain** [HwangCheese/VideoSummary#47]
## What needs refactoring?

The core framework of the TransNetV2 model must be switched from TensorFlow to PyTorch.

**Reasoning:** Using the existing TensorFlow model has led to unstable or complicated local GPU (CUDA) support in newer environments, particularly with TensorFlow versions 2.11 and above. The goal is to migrate to the PyTorch-based `transnetv2_pytorch` module to enable stable, explicit GPU acceleration via an initialization like `model = TransNetV2(device="cuda")`.

---

## Detailed tasks

- [ ] **Identify and Prepare Replacement for Existing TensorFlow Logic**
- Identify the current TensorFlow-based TransNetV2 model loading and inference code that needs to be replaced. Prepare the structure for the PyTorch-based replacement logic.

- [ ] **Remove or simplify unnecessary logic**

- Remove TensorFlow dependencies and implement the new logic using the `transnetv2_pytorch` library.
- Implement the function to accept a `device` argument to explicitly handle GPU/CPU assignment.

- [ ] **Verify functionality remains consistent**
- Verify that the PyTorch model maintains the same detection accuracy as the original TensorFlow model.
- Confirm that GPU acceleration works correctly when `device="cuda"` is specified, and measure the performance gain (speed) compared to CPU usage.

---

## Expected outcome

The TransNetV2 model will be successfully transitioned to the PyTorch (`transnetv2_pytorch`) framework, providing stable CUDA GPU acceleration. Explicit device assignment via `device="cuda"` will be possible, and the scene detection time for large videos will be significantly reduced.

---

## References (optional)

- **Target Module Code Snippet:**
```
# Key Change
from transnetv2_pytorch import TransNetV2 
# ...
model = TransNetV2(device=device) 
scenes = model.detect_scenes(video_path)
```

- **Related Library:** `transnetv2-pytorch`
